### PR TITLE
dird: add Job warning if Admin job contains invalid RunScripts

### PR DESCRIPTION
**Pull request by joergsteffens**
Currently, RunScripts in Admin Jobs configured with "RunsOnClient = Yes" (or no RunsOnClient setting at all, as this is the default) have been silently ignored.
It has been hard, to figure out if the RunScript did run or not, as there has been no hint in the job log.

With this change, an error job message will be generated
and the job will be set to "Admin Error".

### CODE REVIEW

File: core/src/lib/runscript.cc [hunk 1]
```diff
@@ -134,41 +134,49 @@ int RunScripts(JobControlRecord* jcr,
...
-    if ((script->when & SCRIPT_Before) && (when & SCRIPT_Before)) {
-      if ((script->on_success &&
-           (jcr->JobStatus == JS_Running || jcr->JobStatus == JS_Created)) ||
-          (script->on_failure &&
-           (JobCanceled(jcr) || jcr->JobStatus == JS_Differences))) {
-        Dmsg4(200, "runscript: Run it because SCRIPT_Before (%s,%i,%i,%c)\n",
-              script->command.c_str(), script->on_success, script->on_failure,
-              jcr->JobStatus);
-        runit = true;
+    if (!script->IsLocal()) {
```
Comment (franku) [inline core/src/lib/runscript.cc]: As far as I can see, currently, every runscript is by default not local (runscript->target is not empty). As a matter of fact this statement is always true and the following else won't be reached.

File: core/src/lib/runscript.cc [hunk 2]
```diff
@@ -134,41 +134,49 @@ int RunScripts(JobControlRecord* jcr,
...
+             (JobCanceled(jcr) || jcr->JobStatus == JS_Differences))) {
+          Dmsg4(200, "runscript: Run it because SCRIPT_After (%s,%i,%i,%c)\n",
+                script->command.c_str(), script->on_success, script->on_failure,
+                jcr->JobStatus);
+          runit = true;
+        }
       }
     }
 
-    if (!script->IsLocal()) { runit = false; }
```
Comment (franku) [inline core/src/lib/runscript.cc]: As well as the code above removing this line changes the default behaviour.

**Comment by franku** [comment]
As this PR is dependent on #358 and changes default behaviour of the way runscripts are handled I would not accept it without urgent need. Please understand that I cannot accept his change until we have a solution for #358. Thanks for contribution anyway.

**Comment by joergsteffens** [comment]
The current incorrect behavior is also described in https://bugs.bareos.org/view.php?id=419.

**Comment by joergsteffens** [comment]
This PR is not directly related to #358 and in essence only add one code line for a job message warning.

**Comment by joergsteffens** [comment]
Adapted change as internal discussed. Do not set the Admin Job to error, just add a warning to the joblog.

File: core/src/lib/runscript.cc (hunk 1 above)
Comment (joergsteffens) [reply core/src/lib/runscript.cc]: No, as discussed in PR #358.

**Comment by joergsteffens** [comment]
As said before, in essence this change only adds one code line for a job message warning.
The reason why the diff show many modified lines is that the if statement "if (!script->IsLocal())" have been moved from being the last check to being the first check, to better represent the indented logic. Therefore the indentation of the code changed.

File: core/src/lib/runscript.cc (hunk 2 above)
Comment (franku) [reply core/src/lib/runscript.cc]: This moved to the if clause in line 137. -> resolved

**Pull request merged**